routes/shopItem.js

```diff
@@ -32,7 +28,19 @@ shopItemRouter.post("/", (req, res, next) => {
 });
 
 shopItemRouter.put("/", (req, res, next) => {
-
+    if (typeof req.body.id != "number") {
+        res.status(400).send();
+    }
+    const isIdInOurDatabase = req.database.shopItems.findIndex((item) => {
+        return item.id === req.body.id;
+    })
+    if (isIdInOurDatabase !== -1) {
+        req.database.shopItems[isIdInOurDatabase].name = req.body.name;
```

Comment:
The idea of PUT is to "overwrite" the original object with an updated object. It should work similarly to POST but the "new" object must already exist (otherwise PUT should retune 404). The code here does everything right except to add/update `isCompleted` similarly to POST.

Something similar to this would solve the issue.

```suggestion
        if ("isCompleted" in req.body === false) {
            req.body.isCompleted = false;
        }
        req.database.shopItems[isIdInOurDatabase] = req.body;
```